### ISSUE TITLE
fix(inference): implement temperature-scaled sampling (currently greedy-only)

### ISSUE THREAD

**Issue by sankpal-shreyas** [Shreyas582/WraithRun#66]
﻿## Problem

`ModelConfig.temperature` is accepted as a CLI argument and stored in config, but `select_next_token()` always performs **greedy argmax** — no temperature scaling or sampling is applied.

`
ust
fn select_next_token(logits_value: &DynValue) -> Result<i64> {
    // ... always takes argmax, ignores temperature entirely
}
`

The `--temperature` flag and preset profile temperatures (e.g., live-fast=0.1, live-balanced=0.2) have no effect on generation.

## Expected Behavior

When `temperature > 0`:
1. Scale logits by `1.0 / temperature`
2. Apply softmax to get probabilities
3. Sample from the distribution (or use top-k/top-p)

When `temperature == 0` (or very small): keep greedy argmax.

## Scope

- [ ] Pass `temperature` to `select_next_token` / `run_prompt`
- [ ] Implement temperature-scaled sampling
- [ ] Keep greedy mode as default when temperature is 0 or unset
- [ ] Add tests for sampling behavior
